### ISSUE TITLE
installs wrong version of modules

### ISSUE THREAD

**Issue by ColCh** [npm/npm#12654]
npm versions affected as well: `3.8.9`, `3.9.0`

After `npm cache clean`, `rm -r node_modules` and `npm install`, I experience following:
( I have `"lodash": "^4.11.1",` in my deps)

`npm ls`:

```
npm ERR! missing: lodash@^3.10.0, required by babel-core@6.8.0
npm ERR! missing: lodash@^3.10.1, required by babel-generator@6.8.0
npm ERR! missing: lodash@^3.10.0, required by babel-register@6.8.0
npm ERR! missing: lodash@^3.10.1, required by babel-template@6.8.0
npm ERR! missing: lodash@^3.10.1, required by babel-traverse@6.8.0
npm ERR! missing: lodash@^3.10.1, required by babel-types@6.8.1
npm ERR! missing: lodash@^3.10.0, required by babel-plugin-transform-es2015-block-scoping@6.8.0
npm ERR! missing: lodash@^3.10.0, required by babel-helper-define-map@6.8.0
npm ERR! missing: lodash@^3.10.0, required by babel-helper-builder-react-jsx@6.8.0
npm ERR! missing: lodash@^3.2.0, required by babel@5.8.38
npm ERR! missing: lodash@^3.10.0, required by babel-core@5.8.38
npm ERR! missing: lodash@^3.9.3, required by babel-plugin-proto-to-assign@1.0.4
npm ERR! missing: lodash@^3.10.1, required by grouped-queue@0.3.2
npm ERR! missing: lodash@^3.5.0, required by yeoman-generator@0.20.3
npm ERR! missing: lodash@^3.2.0, required by cheerio@0.19.0
npm ERR! missing: lodash@^3.6.0, required by yeoman-assert@2.2.1
npm ERR! missing: lodash@^3.3.1, required by inquirer@0.8.5
```

Running `npm install` second time resolves this problem:

```
➜  FOO git:(navigation) npm install
FOO@0.0.1 /FOO
├─┬ babel-core@6.8.0
│ ├─┬ babel-generator@6.8.0
│ │ └── lodash@3.10.1 
│ ├─┬ babel-register@6.8.0
│ │ └── lodash@3.10.1 
│ ├─┬ babel-template@6.8.0
│ │ └── lodash@3.10.1 
│ ├─┬ babel-traverse@6.8.0
│ │ └── lodash@3.10.1 
│ ├─┬ babel-types@6.8.1
│ │ └── lodash@3.10.1 
│ └── lodash@3.10.1 
└─┬ react-native@0.22.2
  ├─┬ babel-preset-react-native@1.7.0
  │ ├─┬ babel-plugin-transform-es2015-block-scoping@6.8.0
  │ │ └── lodash@3.10.1 
  │ ├─┬ babel-plugin-transform-es2015-classes@6.8.0
  │ │ └─┬ babel-helper-define-map@6.8.0
  │ │   └── lodash@3.10.1 
  │ └─┬ babel-plugin-transform-react-jsx@6.8.0
  │   └─┬ babel-helper-builder-react-jsx@6.8.0
  │     └── lodash@3.10.1 
  ├─┬ fbjs-scripts@0.4.0
  │ └─┬ babel@5.8.38
  │   ├─┬ babel-core@5.8.38
  │   │ └─┬ babel-plugin-proto-to-assign@1.0.4
  │   │   └── lodash@3.10.1 
  │   └── lodash@3.10.1 
  ├─┬ yeoman-environment@1.6.1
  │ └─┬ grouped-queue@0.3.2
  │   └── lodash@3.10.1 
  └─┬ yeoman-generator@0.20.3
    ├─┬ html-wiring@1.2.0
    │ └─┬ cheerio@0.19.0
    │   └── lodash@3.10.1 
    ├── lodash@3.10.1 
    └─┬ yeoman-assert@2.2.1
      └── lodash@3.10.1 
```

But I can't just run `npm install` twice every time :( Also my CI is broken at this point

**Comment by ColCh** [comment]
What ... 

I moved my `"lodash": "^4.11.1"` dependency from `devDependencies` to `dependencies` and now it works fine.

However, this is wrong behaviour, isn't it?

**Comment by lxe** [comment]
This is probably related to #10119

**Comment by lxe** [comment]
This is a major problem with npm 3 and it stuck around for most of its major revisions. The workaround so far has been to either :

 1) Install twice; or
 2) Merge devDependencies into dependencies and then run `npm install`

**Comment by ColCh** [comment]
@lxe It's probably a dupe for #10277, but I'm not sure. I leave issue open, until someone from *npm* team will take a look on this issue.

**Comment by s-a** [comment]
I experiance this behavior with my module. Recently changed from electron 0.x to 1.x. Tips above solve the issue only for current console session.  NPM -v 2.15.1
Ref: https://github.com/s-a/iron-node/issues/104

**Comment by s-a** [comment]
Sorry for the noise. I was confused by circumstances which are related to electron. Please ignore my last post.

**Comment by schmod** [comment]
Looks like a dupe of #10727

**Comment by zkat** [comment]
This is quite probably fixed as of npm@4.0.5 thanks to #12811. Cheers!
